plugins/batterystat.koplugin/main.lua

```diff
@@ -0,0 +1,273 @@
+
+local DataStorage = require("datastorage")
+local KeyValuePage = require("ui/widget/keyvaluepage")
+local LuaSettings = require("luasettings")
+local PowerD = require("device"):getPowerDevice()
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local logger = require("logger")
+local util = require("ffi/util")
+local _ = require("gettext")
+
+local State = {}
+
+function State:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    if o.percentage == nil or o.timestamp == nil then
+        o.percentage = PowerD:getCapacity()
+        o.timestamp = os.time()
+    end
+    return o
+end
+
+function State:toString()
+    return "{ " .. self.percentage .. " @ " .. os.date("%c", self.timestamp) .. "}"
```

Comment:
This is for performance reason:

```lua
string.format("{ %s @ %s }", self.percentage, os.date("%c", self.timestamp)
```

The above only creates 2 strings at runtime, while the string concatenation version will create 4 strings. It's a best practice for any language that does string interning, for example, Python and Lua.